elastic/kibana
# [Stateful sidenav]  User profile opt in/out

As part of the onboarding users to the new "solution" navigation, we want to allow the user to opt out and revert to the current "classic" navigation.

In this PR I've added a toggle in the user profile dropdown to allow reverting to the "classic" navigation.

## How to test

* Add the following setting in `kibana.yml`

```yml
navigation.solutionNavigation.featureOn: true
navigation.solutionNavigation.enabled: true
navigation.solutionNavigation.optInStatus: visible
navigation.solutionNavigation.defaultSolution: es
xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
xpack.cloud.base_url: "https://cloud.elastic.co"
xpack.cloud.deployment_url: "/deployments/deploymentId"
xpack.cloud.organization_url: "/account/"
xpack.cloud.billing_url: "/billing/"
xpack.cloud.profile_url: "/user/settings/"
xpack.cloud.performance_url: "/performance/"
xpack.cloud.users_and_roles_url: "/users-and-roles/"
```

* Lauch Kibana
* There should be a toggle in the user menu to opt out or back in of the new navigation

## Screenthot

<img width="297" alt="Screenshot 2024-03-22 at 16 30 43" src="https://github.com/elastic/kibana/assets/2854616/538d14e6-ac72-4300-9fa4-d43c5f142e66">

## Note on implementation changes

I've also changed the way the project navigation is initiated by requiring an `id` to be passed. This simplified the implementation to detect if we try to initiate a navigation tree multiple times.
